### PR TITLE
fix: Unprotect list detail page

### LINKED ISSUE #1792: The address for the default list is visible but it shouldn't
## 🎉 Description

When getting into the default list view, the zero address `0x0..` is visible due to the default list having that address for everyone.
Hide the address when the list is the default one.

### LINKED ISSUE #1793: The list is view route is protected but it shouldn't
## 🎉 Description

The current list view is protected, meaning that it requires the user to be logged in. Check if we can remove this protection and if the route can be seen without being logged in as well.

### PR DESCRIPTION
**Pull request by kevinszuchet**
Merge after releasing https://github.com/decentraland/marketplace-favorites-server/pull/170
Closes #1793 
Closes #1792 

There is a visual bug when signing out because of a re-rendering.
https://www.loom.com/share/b8e3bc2dbb414c939e655f9abf6da1b4?sid=e745f041-71c1-4a5b-89cd-ee4ca94d8766

### CODE REVIEW

**Review by flobarreto — APPROVED** [review]

File: webapp/src/modules/favorites/sagas.spec.ts [hunk 1]
```diff
@@ -305,125 +304,190 @@ describe('when handling the request for fetching favorited items', () => {
...
-            })
-            .put(fetchFavoritedItemsFailure(error.message))
-            .dispatch(fetchFavoritedItemsRequest(options))
-            .run({ silenceTimeout: true })
+    describe('and the call to the favorites api succeeds', () => {
+      let favoritedItemIds: FavoritedItems
+      let createdAt: Record<string, number>
+      let total: number
+
+      describe("and there's more than favorited item", () => {
```
Comment (LautaroPetaccio) [inline webapp/src/modules/favorites/sagas.spec.ts:362]:
```suggestion
      describe("and there's more than one favorited item", () => {
```

File: webapp/src/modules/favorites/sagas.spec.ts [hunk 2]
```diff
@@ -432,51 +496,169 @@ describe('when handling the request for fetching favorited items', () => {
...
           .run({ silenceTimeout: true })
       })
     })
+
+    describe('and the call to the favorites api succeeds', () => {
+      let favoritedItemIds: FavoritedItems
+      let createdAt: Record<string, number>
+      let total: number
+
+      describe("and there's more than favorited item", () => {
```
Comment (LautaroPetaccio) [inline webapp/src/modules/favorites/sagas.spec.ts:528]:
```suggestion
      describe("and there's more than one favorited item", () => {
```

File: webapp/src/components/ListPage/ListPage.tsx [hunk 3]
```diff
@@ -126,6 +129,10 @@ const ListPage = ({
     fetchList()
   }, [fetchList])
 
+  if (!isConnecting && !wallet && list?.isPrivate) {
```
Comment (LautaroPetaccio) [inline webapp/src/components/ListPage/ListPage.tsx:132]: Can a non-logged in user receive a list with `isPrivate` in true?

**Review by LautaroPetaccio — COMMENTED** [review]
The PR looks great! I've left a few typos and a comment to review.

**Review by LautaroPetaccio — APPROVED** [review]
🚢

**Pull request merged**